java/src/main/java/ai/rapids/cudf/HostMemoryBuffer.java

```diff
@@ -59,9 +78,11 @@ protected boolean cleanImpl(boolean logErrorIfNotClean) {
   }
 
   /**
-   * Factory method to create this buffer
-   * @param bytes - size in bytes to allocate
-   * @return - return this newly created buffer
+   * Allocate memory, but be sure to close the returned buffer to avoid memory leaks.
+   * @param bytes size in bytes to allocate
+   * @param preferPinned true if the pinned memory pool should be used, else false to fall
+   *                     back to using regular off heap memory.
```

Comment:
I think the semantics need to be clarified a bit here, as one may interpret a `true` parameter to mean it will _only_ be pinned memory.  How about something like,
```
If set to true, the pinned memory pool will be used if possible with a fallback to off-heap memory.  If set to false, the allocation will always be from off-heap memory.
```